stylix/testbed.nix

```
@@ -85,10 +127,26 @@ let
         # Disable the GNOME tutorial which pops up on first login.
         environment.gnome.excludePackages = [ pkgs.gnome-tour ];
 
+        # for use when application is set
         environment.systemPackages = [
-          (pkgs.makeAutostartItem {
-            inherit (config.stylix.testbed.application) name package;
-          })
+          (pkgs.makeAutostartItem (
+            if (config.stylix.testbed.ui.application != null) then
+              {
+                inherit (config.stylix.testbed.ui.application) name package;
+              }
+            else if (config.stylix.testbed.ui.command != null) then
```

This if-else construct implies that only one of them has an effect, with `config.stylix.testbed.ui.application` being prioritized. Either both being non-`null` should cause their consequences to be merged, or we explicitly disallow this with an

```nix
!(
  config.stylix.testbed.ui.application != null &&
  config.stylix.testbed.ui.command != null
)
```

assertion to prevent silently ignoring `config.stylix.testbed.ui.command` declarations.

Although merging the consequences is the most extensible, simply adding an assertion is also fine since we currently do not need this functionality. I leave the decision up to you.

> Just having one option which passes an attrset directly to `makeAutostartItem` would require less code (unless I've overlooked something), and potentially support more use cases

Simply delegating `pkgs.makeAutostartItem` arguments is also an option, especially since the `config.stylix.testbed.{command,ui.application}` distinction is not directly apparent. Otherwise, the names could be somehow improved.

---

> > Just having one option which passes an attrset directly to `makeAutostartItem` would require less code (unless I've overlooked something), and potentially support more use cases
> 
> Simply delegating `pkgs.makeAutostartItem` arguments is also an option, especially since the `config.stylix.testbed.{command,ui.application}` distinction is not directly apparent. Otherwise, the names could be somehow improved.

Small correction: `config.stylix.testbed.{command,ui.application}` is actually `config.stylix.testbed.ui.{command,application}`, which may make their distinction slightly clearer (or maybe less clear), though I agree that it is hard to immediately understand the difference from their names. I am not sure what names would better differentiate them, though, and I believe that their option descriptions do a fair job of explaining what each does.

From what i can grasp about `pkgs.makeAutoStartItem` from its [definition](https://github.com/hsjobeki/nixpkgs/blob/master/pkgs/build-support/make-startupitem/default.nix), it would require [this set](https://github.com/Flameopathic/stylix/blob/623b022cddc3c549f47c56a466ab81ddf83cf9e2/stylix/testbed.nix#L138-L146) be duplicated with only one or to arguments being changed across every application testbed which runs an arbitrary command.

Also, doesn't the current [upstream implementation](https://github.com/danth/stylix/blob/8d5cd725ad591890c0cd804bf68cc842b8afca51/stylix/testbed.nix#L91-L93) already basically just pass the set directly to `makeAutoStartItem`?